returns/maybe.py

```diff
@@ -430,10 +436,10 @@ def failure(self):
         raise UnwrapFailedError(self)
 
     def filter(self, function):
```

Comment:
Is it possible to create a common `filter_` function for all possible types? `Result`, `IOResult`, ...?

Reply:
`filter` is special in the sense that it 'empties' the container, which is possible in `Maybe`, but not in `Result` (i.e. what would the failure value of `Success(4).filter(lambda x: False)` be?)

Reply:
Ok, I see! Than, it looks like we can use `MaybeLikeN` to limit the input type of `filter_`?

Reply:
@sobolevn I used it, please tell me if its something you were looking for. I am not entirely sure if I did it right :)